projects/js-packages/publicize-components/src/components/social-image-generator/template-picker/index.js

```diff
@@ -0,0 +1,108 @@
+import { ThemeProvider } from '@automattic/jetpack-components';
+import { Button, Modal } from '@wordpress/components';
+import { useState, useCallback, useMemo } from '@wordpress/element';
+import { sprintf, __ } from '@wordpress/i18n';
+import classnames from 'classnames';
+import React from 'react';
+// Template thumbnails
+import Dois from '../assets/dois.jpg';
+import Edge from '../assets/edge.jpg';
+import Fullscreen from '../assets/fullscreen.jpg';
+import Highway from '../assets/highway.jpg';
+import styles from './styles.module.scss';
+
+const TEMPLATES_DATA = [
+	{
+		name: 'highway',
+		label: 'Highway',
+		image: Highway,
+	},
+	{
+		name: 'dois',
+		label: 'Dois',
+		image: Dois,
+	},
+	{
+		name: 'edge',
+		label: 'Edge',
+		image: Edge,
+	},
+	{
+		name: 'fullscreen',
+		label: 'Fullscreen',
+		image: Fullscreen,
+	},
+];
+
+/**
+ * A component that displays a modal with a list of templates to choose from.
+ *
+ * @param {object} props - The component props.
+ * @param {Function} props.onSelect - A function that will be called when a template is selected. Receives the name of the selected template as an argument.
+ * @param {Function} props.render - A function that will be called with an object containing an "open" function, which can be called to open the template picker.
+ * @param {string|null} [props.value=null] - The name of the currently selected template.
+ * @returns {React.ReactNode} - The component's rendered output.
+ */
+const TemplatePicker = ( { onSelect, render, value = null } ) => {
+	const [ isOpen, setIsOpen ] = useState( false );
+	const [ selectedTemplate, setSelectedTemplate ] = useState( value );
+
+	const openPicker = useCallback( () => setIsOpen( true ), [ setIsOpen ] );
+	const closePicker = useCallback( () => {
+		setSelectedTemplate( value );
+		setIsOpen( false );
+	}, [ value, setSelectedTemplate, setIsOpen ] );
+	const saveAndClosePicker = useCallback( () => {
+		onSelect( selectedTemplate );
+		setIsOpen( false );
+	}, [ onSelect, setIsOpen, selectedTemplate ] );
+
+	// Add memoized callback function for each template.
+	const TEMPLATES = useMemo(
+		() =>
+			TEMPLATES_DATA.map( template => ( {
+				...template,
+				onSelect: () => setSelectedTemplate( template.name ),
+			} ) ),
+		[ setSelectedTemplate ]
+	);
+
+	return (
+		<ThemeProvider targetDom={ document.body }>
```

Comment:
The `ThemeProvider` component is required to get access to Jetpack's RNA styles. We might consider wrapping the entire Jetpack Social sidebar in it instead.